# Tarea 02b

Utilizamos Python como calculadora. Para hacer la cosa más interesante los ejercicios de este *Notebook* deben ser resueltos utilizando **únicamente** los operadores aritméticos `+ - / * ** %` y `//` operando entre literales y/o las funciones de la librería estándard (.

## Ejercicio 1

Calcular el módulo de la aceleración que sufre el electrón en el estado fundamental del átomo de hidrógeno. Expresar el resultado en unidades SI.

Vuelta de tuerca: Estimar el error en la aceleración calculada utilizando un modelo de propagación lineal de errores y asumiendo que los errores en las cantidades necesarias para el cálculo no están correlacionados.

Utilizar los valores de las constantes físicas de [aquí](https://physics.nist.gov/cuu/Constants/index.html), hay info útil [aquí](https://arxiv.org/pdf/1610.08716.pdf) y [aquí](https://xkcd.com/2110/).

### Cálculo Modulo aceleración de un electrón en el estado fundamental del átomo de hidrógeno
---
1. Buscamos las constantes necesarias para el cálculo por un potencial Coulombiano de un protón y un electrón, donde la distancia corresponde al radio de bohr 

>Physical constants & standard uncertainty in parenthesis 
>
>R_bohr $a_{0}$: 0.529 177 210 903(80)×10−10m    
>Electon mass $m_{e}$: 9.109 383 7015(28) x 10-31 kg     
>Elementary charge $e$: 1.602 176 634(0) x 10-19 C    
>vacuum electric permittivity $\varepsilon_0$:  8.854 187 8128(13) x 10-12 F m-1   
> $\pi$: 3.1415926536 (aprox al 10mo decimal)
         3.141 592 653 589 793 238

2. Por ley de Coulomb tenemos que el modulo de la fuerza entre 2 partículas con carga:

> $$|\overrightarrow{F}|= \frac{1}{4\pi \epsilon_{0}} \frac{|q_{e}||q_{p}|}{r^2} $$

3. Y la aceleración por ley de newton es:

>$$ a = |\overrightarrow{a}| = \frac{|\overrightarrow{F}|}{m_{e}}$$

4. Tenemos finalmente que la aceleración en función de las cargas y la distancia esta dada por:

>$$a = |\overrightarrow{a}| = \frac{1}{4\pi \epsilon_{0}} \frac{|q_{e}||q_{p}|}{r^2} \frac{1}{m_{e}} $$

In [15]:
pow(1.602_176_634e-19,2) / (4 * 3.1415926536 * 8.854_187_8128e-12 * pow(0.529_177_210_903e-10,2) * 9.109_383_7015e-31) 


9.04421612721529e+22

Para la propagación de errores, asumiendo que las variables no están correlacionadas tenemos entonces que el error final estará dado por las variables que tenemos radio $r$, valores de las cargas $q_{e}$ y $q_{p}$, y de la masa del electrón $m_{e}$:
>
>$$\sigma_{a} = \sqrt{\left( \frac{\partial a}{\partial r} \sigma_{r} \right) ^2 + \left( \frac{\partial a}{\partial q_{e}} \sigma_{q_{e}} \right) ^2 + \left( \frac{\partial a}{\partial q_{p}} \sigma_{q_{p}} \right) ^2 + \left( \frac{\partial a}{\partial m_{e}} \sigma_{m_{e}} \right) ^2 + \left( \frac{\partial a}{\partial \epsilon_{0}} \sigma_{\epsilon_{0}} \right) ^2}$$

Calculemos las derivadas parciales por separado para mantener el orden:

>La incertidumbre en la carga es cero según la última redefiinición de las unidades SI, por lo tanto los terminos correspondiente a carga de protón y electrón se anulan, nos queda entonces 
>
>$$\frac{\partial a}{\partial r} =  -\frac{|q_{e}||q_{p}|}{2 \pi \epsilon_{0} m_{e}} \frac{1}{r^3} $$
>
>$$\sigma_{r} = 0.00000000080 x 10^{-11} m $$

Evaluamos ahora esta primera derivada respecto al radio o distancia entre las cargas:

In [ ]:
- pow(1.602_176_634e-19, 2) / (2 * 3.1415926536 * 8.854_187_8128e-12 * 9.109_383_7015e-31 * pow(0.529_177_210_903e-10, 3))


-3.418218298472088e+33

Seguimos con la derivada respecto a la masa del electrón:
>$$\frac{\partial a}{\partial m_{e}} = -\frac{|q_{e}||q_{p}|}{4 \pi \epsilon_{0}r^{2}}  \frac{1}{m_{e}^2} $$ 
>
>$$\sigma_{m_{e}} =  0 .000 000 0028 x 10^{-31} kg$$
> 

In [ ]:
- pow(1.602_176_634e-19, 2) / (4 * 3.1415926536 * 8.854_187_8128e-12 * pow(0.529_177_210_903e-10,2) * pow(9.109_383_7015e-31, 2))


-9.928461050253068e+52

Finalmente la derivada respecto a la permitividad eléctrica: 

>$$\frac{\partial a}{\partial m_{e}} = -\frac{|q_{e}||q_{p}|}{4 \pi r^{2} m_{e}}  \frac{1}{\epsilon_{0}^2} $$ 
>
>$$\sigma_{epsilon_{0}} =   0.0000000013 x 10^{-12} F/m$$
>

In [ ]:
- pow(1.602_176_634e-19, 2) / (4 * 3.1415926536 * pow(0.529_177_210_903e-10,2) * 9.109_383_7015e-31 * pow(8.854_187_8128e-12, 2))


-1.0214619701358238e+34

Finalmente el error en la aceleración será:

>
>$$\sigma_{a} = \sqrt{\left( \frac{\partial a}{\partial r} \sigma_{r} \right) ^2 + \left( \frac{\partial a}{\partial m_{e}} \sigma_{m_{e}} \right) ^2 + \left( \frac{\partial a}{\partial \epsilon_{0}} \sigma_{\epsilon_{0}} \right) ^2}$$
>
Reemplazando valores obtenemos el valor final:

In [ ]:
pow(pow(-3.418218298472088e+33 * 0.00000000080e-11,2) + pow(-9.928461050253068e+52 * 0.0000000028e-31,2) + pow(-1.0214619701358238e+34 * 0.0000000013e-12,2),1/2) 

41193988055793.086

Tenemos entonces que el valor final de la aceleración es:

$$a = 9.04421612721529 \ \pm 0.0000000041194 \ \small x \normalsize \ 10^{22} \ m/s^{2}$$


## Ejercicio 2

Calcular una aproximación para el Número de Euler ($e$). Hay una buena lista de formas para hacerlo en [Wikipedia](https://en.wikipedia.org/wiki/E_(mathematical_constant)).

### Cálculo aproximado del número de Euler($e$)
---

La manera más fácil y rápida de calcular el número de Euler es mediante su representación como serie infinita:

>$$ e =\sum^{\infty}_{n=0} \frac{1}{n!} = \frac{1}{1} +\frac{1}{1} +\frac{1}{1 \cdot 2}+\frac{1}{1 \cdot 2 \cdot 3} + \frac{1}{1 \cdot 2 \cdot 3 \cdot 4} + \frac{1}{1 \cdot 2 \cdot 3 \cdot 4 \cdot 5} + \cdots $$

Podemos tomar tantos terminos como tan buena queremos nuestra aproximación, hasta 50 decimales   
e = 2.71828182845904523536028747135266249775724709369995... 

A continuación tomamos los 10 primeros términos (desde n=0 hasta n=9) y nos da:

In [ ]:
(1 / 1) + (1 / 1) + (1 / (1 * 2)) + (1 / (1 * 2 * 3)) + (1 / (1 * 2 * 3 * 4)) + (1 / (1 * 2 * 3 * 4 * 5)) + (1 / (1 * 2 * 3 * 4 * 5 * 6)) + (1 / (1 * 2 * 3 * 4 * 5 * 6 * 7)) + (1 / (1 * 2 * 3 * 4 * 5 * 6 * 7 * 8)) + (1 / (1 * 2 * 3 * 4 * 5 * 6 * 7 * 8 * 9
                                                                                                                                                                                                                             ))

2.7182815255731922

## Ejercicio 3

Utilizando el resultado anterior calcular valores aproximados para $\cos(1/2)$ y $\sin(1/3)$. Estimar los errores para cada valor.

##Cálculo de cos(1/2) y sin (1/3)
--
Podemos representar tanto el seno como el coseno de un valor en base a la suma o resta de exponenciales complejas de la siguiente manera:
>$$sen(x)= \frac{e^{ix} - e^{-ix}}{2i}$$
>
>$$cos(x)= \frac{e^{ix} + e^{-ix}}{2}$$

Podemos calcular tanto el sen como el cos utilizando el valor de e calculado en el ejercicio anterior y como el resultado será un valor x de tipo "complex"  con x.real podemos extraer el valor buscado.

Entonces para el calculo de sen(1/3) tenemos:


In [ ]:
(((2.7182815255731922 ** (1j * (1 / 3))) - (2.7182815255731922 ** (-1j * (1 / 3)))) / (2 * 1j)).real

0.32719466169872347

Y para el cos(1/2):


In [ ]:
(((2.7182815255731922 ** (1j * (1 / 2))) + (2.7182815255731922 ** (-1j * (1 / 2)))) / 2).real

0.8775825886004828

## Ejercicio 4

Y si queremos $\tan(\sqrt{2}/2)$?

## Cálculo $tan(\sqrt{2}/ 2)$
---
Para este cálculo podemos aprovechar la manera de operar de los ejercicios anteriores para calcular el $sen(\sqrt{2}/ 2)$ y el $cos(\sqrt{2}/ 2)$  y obtener el resultado final como el cociente de estos dos ya que por definición:

> $$tan(\sqrt{2}/ 2) = \frac{sen(\sqrt{2}/ 2)}{cos(\sqrt{2}/ 2)}$$

Tenemos entonces que :


In [ ]:
((((2.7182815255731922 ** (1j * ( pow(2,1/2) / 2))) - (2.7182815255731922 ** (-1j * ( pow(2,1/2) / 2|)))) / (2 * 1j)).real) / ((((2.7182815255731922 ** (1j * ( pow(2,1/2)/ 2))) + (2.7182815255731922 ** (-1j * ( pow(2,1/2) / 2)))) / 2).real  ) 

0.8545102956885845

Subir los ejercicios completos en formato `ipynb` a Google Classroom. Describiendo en celdas de *Markdown* cada paso de los cálculos, sin demasiada exigencia, sólo para que el que lee pueda entender de qué se trata lo que el código hace.